core/aws_ddk_core/cicd/pipeline.py

```diff
@@ -335,6 +342,43 @@ def add_checks(self) -> "CICDPipelineStack":
             self.add_test_stage()
         return self
 
+    def add_custom_stage(self, stage_name: str, steps: List[Dict[str, Any]]) -> "CICDPipelineStack":
+        """
+        Add custom stage to the pipeline.
+
+        Parameters
+        ----------
+        stage_name: str
+            Name of the stage
+        steps: List[Dict[str, List[str], Optional[IFileSetProducer], Optional[List[str]]]]
+            Steps to add to this stage. Should be a dictionary with the follwing keys set.
+            - name: str
+            - commands: List[str]
+            - input: Optional[IFileSetProducer]
+            - install_commands: Optional[List[str]]
+            See `Documentation on aws_cdk.pipelines.ShellStep`
+            <https://docs.aws.amazon.com/cdk/api/v1/python/aws_cdk.pipelines/ShellStep.html>`_ for more detail.
+
+        Returns
+        -------
+        pipeline : CICDPipeline
+            CICD pipeline
+        """
+
+        self._pipeline.add_wave(
+            stage_name,
+            post=[
+                ShellStep(
```

Comment:
I was imagining this a bit differently where the user would pass a list of CDK `ShellSteps` instead of a dictionary with the values. At which point it would simply be:
```
post=steps,
```

The advantage being that if in the future the `ShellStep` definition was to change (e.g. a new argument), we don't need to make any change on our end